runtime/onert/core/src/compiler/train/TensorRegistries.h

```diff
@@ -93,6 +95,15 @@ class TensorRegistries
     return nullptr;
   }
 
+  void iterateTrainableTensors(
+    const std::function<void(const ir::OperandIndex &, backend::train::ITrainableTensor *)> &fn)
+    const
+  {
+    assert(_tensor_regs.size() == 2); // training backend and built-in backcend
+    for (auto &&tensor_reg : _tensor_regs)
```

Comment:
@ragmani I used the same form of `getBackPropITensor` and `getITensor` in the same file. It would be good to replace all patterns in this file at once.

Reply:
I didn't find what form you mentioned. `getBackPropITensor` and `getITensor` are not constant method, and they don't have _range-based for loop_.

Reply:
@ragmani 

> I didn't find what form you mentioned.

Please expand the folding in the same file.
Then, scroll a few lines. You can see the range-based for loop (e.g. line 78, line 89)

```cc
  backend::ITensor *getITensor(ir::OperandIndex index) const
  {
    for (auto &&tensor_reg : _tensor_regs)
    {
      auto tensor = tensor_reg->getITensor(index);
      if (tensor)
        return tensor;
    }
    return nullptr;
  }
```

> `getBackPropITensor` and `getITensor` are not constant method,

What do you mean they are not constant method? 

```cc
  backend::ITensor *getITensor(ir::OperandIndex index) const
```
```cc
  backend::ITensor *getITensor(ir::OperandIndex index) const
```

Reply:
> Please expand the folding in the same file.
Then, scroll a few lines. You can see the range-based for loop (e.g. line 78, line 89)

Thanks, I find that.

> What do you mean they are not constant method?

They are constant method I mentioned. I was looking for `getBackPropITensor` and `getITensor` in the other file. It would be to apply `const auto &` to them too as you mentioned.